packages/mask/src/plugins/Referral/SNSAdaptor/AdjustFarmRewards.tsx

```diff
@@ -0,0 +1,367 @@
+        [farm, chainId, rewardToken],
+    )
+
+    const onChangeTotalFarmReward = useCallback((e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        const totalFarmReward = e.currentTarget.value
+        const attraceFee = Number.parseFloat(totalFarmReward) * (ATTRACE_FEE_PERCENT / 100)
+
+        setTotalFarmReward(totalFarmReward)
+        setAttraceFee(attraceFee)
+    }, [])
+
+    const onChangePageToAdjustRewards = useCallback(() => {
+        props?.onChangePage?.(PagesType.ADJUST_REWARDS, PagesType.ADJUST_REWARDS, {
+            adjustFarmDialog: {
+                farm,
+                rewardToken,
+                referredToken,
+                continue: () => {},
+            },
+        })
+    }, [props?.onChangePage, farm, rewardToken, referredToken])
+
+    const onErrorDeposit = useCallback(
+        (error?: string) => {
+            error && showSnackbar(error, { variant: 'error' })
+            onChangePageToAdjustRewards()
+        },
+        [props?.onChangePage, showSnackbar, onChangePageToAdjustRewards],
+    )
+    const onConfirmedAdjustFarm = useCallback(
+        (txHash: string) => {
+            props?.onChangePage?.(PagesType.TRANSACTION, t.transaction(), {
+                hideAttrLogo: true,
+                hideBackBtn: true,
+                transactionDialog: {
+                    transaction: {
+                        status: TransactionStatus.CONFIRMED,
+                        actionButton: {
+                            label: t.dismiss(),
+                            onClick: onChangePageToAdjustRewards,
+                        },
+                        transactionHash: txHash,
+                    },
+                },
+            })
+        },
+        [props?.onChangePage, t, onChangePageToAdjustRewards],
+    )
+
+    const onConfirmAdjustFarm = useCallback(() => {
+        const { title, subtitle } = getTransactionTitles(
+            Number.parseFloat(totalFarmReward),
+            Number.parseFloat(dailyFarmReward),
+            attraceFee,
+            props.rewardToken,
+        )
+
+        props?.onChangePage?.(PagesType.TRANSACTION, t.transaction(), {
+            hideAttrLogo: true,
+            hideBackBtn: true,
+            transactionDialog: {
+                transaction: {
+                    status: TransactionStatus.CONFIRMATION,
+                    title,
+                    subtitle,
+                },
+            },
+        })
+    }, [props, totalFarmReward, dailyFarmReward, attraceFee])
+
+    const onAdjustFarmReward = useCallback(async () => {
+        if (!referredToken || !rewardToken) {
+            return onErrorDeposit(t.go_wrong())
+        }
+
+        const depositValue = Number.parseFloat(totalFarmReward) + attraceFee
+
+        adjustFarmRewards(
+            onConfirmAdjustFarm,
+            onErrorDeposit,
+            onConfirmedAdjustFarm,
+            web3 as Web3,
+            account,
+            chainId as ChainId,
```

Comment:
```suggestion
            web3,
            account,
            chainId,
```

Reply:
It doesn’t work 2 days ago, I guess you fixed it.